File: .net 6.0/ListenAI.LLM/Models/Request/SparkMessage.cs
using Newtonsoft.Json;

namespace ListenAI.LLM.Models; 

public class SparkMessage {
    /// <summary>
    /// [必填] 角色 (user = 用户的提问, assistant = AI 的回复)
    /// </summary>
    [JsonIgnore]
    public SparkMessageRole Role { get; set; }

    [JsonProperty("role")]
    private string RoleRaw {
        get => Role.Value;
        set {
            Role = new SparkMessageRole(value);
        }
    }

    /// <summary>
    /// [必填] 该角色的对话内容
    /// </summary>
    [JsonProperty("content")]
    public string Content { get; set; }
}

File: java 17/llm/src/main/java/com/listenai/llm/models/request/SparkMessage.java
package com.listenai.llm.models.request;

import com.alibaba.fastjson2.annotation.JSONField;

public class SparkMessage {
    public SparkMessage() {}

    public SparkMessage(String role, String content) {
        this.role = role;
        this.content = content;
    }

    @JSONField(name = "role")
    private String role;

    @JSONField(name = "content")
    private String content;

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }
}
